libraries/AP_BoardConfig/px4_drivers.cpp

```diff
@@ -454,9 +491,17 @@ void AP_BoardConfig::px4_start_optional_sensors(void)
 void AP_BoardConfig::px4_setup_drivers(void)
 {
     px4_start_common_sensors();
-    px4_start_fmuv1_sensors();
-    px4_start_fmuv2_sensors();
-    px4_start_fmuv4_sensors();
+    switch ((px4_board_type)px4.board_type.get()) {
+    case PX4_BOARD_PH2SLIM:
+        px4_start_pixhawk2slim_sensors();
+        break;
+    case PX4_BOARD_AUTO:
+    default:
```

Comment:
Why not add the other cases? Why are we setting the board_type automatically but then don't use it?